From 9ddd1cac6fe1f8464f54ab0d5af9bc8260caca12 Mon Sep 17 00:00:00 2001
Message-Id: <9ddd1cac6fe1f8464f54ab0d5af9bc8260caca12.1651808206.git.geliang.tang@suse.com>
From: Geliang Tang <geliang.tang@suse.com>
Date: Wed, 2 Feb 2022 17:03:40 -0800
Subject: [PATCH] mptcp: print out reset infos of MP_RST
Git-commit: 9ddd1cac6fe1f8464f54ab0d5af9bc8260caca12
Patch-mainline: v5.18-rc1
References: bsn#2
Modified-by-SEL: No


This patch printed out the reset infos, reset_transient and reset_reason,
of MP_RST in mptcp_parse_option() to show that MP_RST is received.

Signed-off-by: Geliang Tang <geliang.tang@suse.com>
Signed-off-by: Mat Martineau <mathew.j.martineau@linux.intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 net/mptcp/options.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/mptcp/options.c b/net/mptcp/options.c
index 7345f28f3de1..3e82ac24d548 100644
--- a/net/mptcp/options.c
+++ b/net/mptcp/options.c
@@ -336,6 +336,8 @@ static void mptcp_parse_option(const struct sk_buff *skb,
 		flags = *ptr++;
 		mp_opt->reset_transient = flags & MPTCP_RST_TRANSIENT;
 		mp_opt->reset_reason = *ptr;
+		pr_debug("MP_RST: transient=%u reason=%u",
+			 mp_opt->reset_transient, mp_opt->reset_reason);
 		break;
 
 	case MPTCPOPT_MP_FAIL:
-- 
2.34.1

